src/main/java/teammates/logic/core/GoogleCloudLoggingService.java

```diff
@@ -94,4 +113,127 @@
+        LogSearchParams logSearchParams = new LogSearchParams()
+                .setLogName(FEEDBACK_SESSION_LOG_NAME)
+                .addLabel(FEEDBACK_SESSION_LOG_COURSE_ID_LABEL, courseId)
+                .addLabel(FEEDBACK_SESSION_LOG_EMAIL_LABEL, email)
+                .setStartTime(startTime)
+                .setEndTime(endTime);
+        List<LogEntry> logEntries = getLogEntries(logSearchParams);
+
+        List<FeedbackSessionLogEntry> fsLogEntries = new ArrayList<>();
+        for (LogEntry entry : logEntries) {
+            String fslType = entry.getLabels().get(FEEDBACK_SESSION_LOG_TYPE_LABEL);
+            long timestamp = entry.getTimestamp();
+            String entryEmail = entry.getLabels().get(FEEDBACK_SESSION_LOG_EMAIL_LABEL);
+            String entryFsName = entry.getLabels().get(FEEDBACK_SESSION_LOG_NAME_LABEL);
+            StudentAttributes student = studentsLogic.getStudentForEmail(courseId, entryEmail);
+            FeedbackSessionAttributes fs = fsLogic.getFeedbackSession(entryFsName, courseId);
+            if (student == null || fs == null) {
+                // If the student email or feedback session retrieved from the logs are invalid, discard it
+                continue;
+            }
+            if (!fslType.equals(Const.FeedbackSessionLogTypes.ACCESS)
+                    && !fslType.equals(Const.FeedbackSessionLogTypes.SUBMISSION)) {
+                // If the feedback session log type retrieved from the logs is invalid, discard it
+                continue;
+            }
+            FeedbackSessionLogEntry fslEntry = new FeedbackSessionLogEntry(student, fs, fslType, timestamp);
+            fsLogEntries.add(fslEntry);
+        }
+
+        return fsLogEntries;
+    }
+
+    private List<LogEntry> getLogEntries(LogSearchParams s) throws LogServiceException {
+        List<LogEntry> logEntries = new ArrayList<>();
+        LoggingOptions options = LoggingOptions.getDefaultInstance();
+
+        List<String> logFilters = new ArrayList<>();
+        if (s.logName != null) {
+            logFilters.add("logName=\"projects/" + options.getProjectId() + "/logs/" + s.logName + "\"");
+        }
+        if (s.startTime != null) {
+            logFilters.add("timestamp>\"" + s.startTime.toString() + "\"");
+        }
+        if (s.endTime != null) {
+            logFilters.add("timestamp<=\"" + s.endTime.toString() + "\"");
+        }
+        for (Map.Entry<String, String> entry : s.labels.entrySet()) {
+            logFilters.add("labels." + entry.getKey() + "=\"" + entry.getValue() + "\"");
+        }
+        String logFilter = logFilters.stream().collect(Collectors.joining("\n"));
+
+        try (Logging logging = LoggingOptions.getDefaultInstance().getService()) {
+            Page<LogEntry> entries = logging.listLogEntries(EntryListOption.filter(logFilter));
+            for (LogEntry entry : entries.iterateAll()) {
+                logEntries.add(entry);
+            }
+            logging.close();
```

Comment:
You should not need this. The fact that `logging` is wrapped within try-with-resources tells that this will be automatically done.

Reply:
Thanks, removed